### ISSUE TITLE
intersection-observer/same-origin-grand-child-iframe.sub.html incorrectly interprets spec

### ISSUE THREAD

**Issue by szager-chromium** [web-platform-tests/wpt#20793]
Cross-posting from: https://github.com/w3c/IntersectionObserver/issues/409

The existing InteresectionObserver spec uses obsolete language in defining [rootBounds](https://w3c.github.io/IntersectionObserver/#dom-intersectionobserverentry-rootbounds):

"If target belongs to the same unit of related similar-origin browsing contexts as the intersection root, this will be the root intersection rectangle. Otherwise, this will be null."

The html spec has deprecated "unit of related similar-origin browsing contexts", but it previously referred to a connected graph of same-origin browsing contexts. The closest concept in the currect spec is probably the "browsing context scope origin":

[https://html.spec.whatwg.org/multipage/browsers.html#browsing-context-scope-origin](https://html.spec.whatwg.org/multipage/browsers.html#browsing-context-scope-origin)

That basically describes any browsing context that is in the "unit of related similar-origin browsing contexts" as the top-level document.

The intent of the original IntersectionObserver spec language was that rootBounds should be null whenever the target is in a cross-origin iframe, even if the top document is same-origin to the target's document.

A recent WPT commit added a test which contradicts this interpretation:

[web-platform-tests/wpt@57d1b42](https://github.com/web-platform-tests/wpt/commit/57d1b42674b8226de887e68e8e73bfd9257fb333)

I propose that the test be removed from WPT.

**Comment by szager-chromium** [comment]
I've now managed to un-convince myself of my initial analysis; but I'm not actually sure what the final disposition should be, so I would appreciate input...

**Comment by hiikezoe** [comment]
Yeah, it shouldn't be there until the spec clearly describes the concept of the similar origin browsing contexts.
Though I don't have strong opinion on the concept, I tend to agree with you.
